enclave-runtime/src/rpc/worker_api_direct.rs

```diff
@@ -293,6 +348,142 @@ fn attesteer_forward_ias_attestation_report_inner(
 	Ok(ext)
 }
 
+fn personhoodoracle_parse_params(
+	params: Params,
+) -> Result<(CommunityIdentifier, CeremonyIndexType, AccountId, CeremonyIndexType), String> {
+	let hex_encoded_params = params.parse::<Vec<String>>().map_err(|e| format!("{:?}", e))?;
+
+	if hex_encoded_params.len() != 4 {
+		return Err(format!(
+			"Wrong number of arguments for fetch_reputation_inner: {}, expected: {}",
+			hex_encoded_params.len(),
+			1
+		))
+	}
+
+	let cid = itp_utils::hex::decode_hex(&hex_encoded_params[0]).map_err(|e| format!("{:?}", e))?;
+	let cid: CommunityIdentifier =
+		Decode::decode(&mut cid.as_slice()).map_err(|e| format!("{:?}", e))?;
+
+	let cindex =
+		itp_utils::hex::decode_hex(&hex_encoded_params[1]).map_err(|e| format!("{:?}", e))?;
+	let cindex: CeremonyIndexType =
+		Decode::decode(&mut cindex.as_slice()).map_err(|e| format!("{:?}", e))?;
+
+	let account =
+		itp_utils::hex::decode_hex(&hex_encoded_params[2]).map_err(|e| format!("{:?}", e))?;
+	let account: AccountId =
+		Decode::decode(&mut account.as_slice()).map_err(|e| format!("{:?}", e))?;
+
+	let number_of_reputations =
+		itp_utils::hex::decode_hex(&hex_encoded_params[3]).map_err(|e| format!("{:?}", e))?;
+	let number_of_reputations: CeremonyIndexType =
+		Decode::decode(&mut number_of_reputations.as_slice()).map_err(|e| format!("{:?}", e))?;
```

Comment:
So you have not tried to use use the serialization? What was the problem with the decoding then?

Reply:
Alright, fine with me. 👍